### ISSUE TITLE
how to cast types?

### ISSUE THREAD

**Issue by pthieu** [xenova/transformers.js#557]
### Question

I have the following code:

```
const pipe = await pipeline('embeddings');
      const output = await pipe([
        'The quick brown fox jumps over the lazy dog',
      ]);
      const embedding = output[0][0];
```

`output[0][0]` causes a typescript error：
<img width="748" alt="CleanShot 2024-02-01 at 23 38 04@2x" src="https://github.com/xenova/transformers.js/assets/2908721/6e7a1e58-bfbf-4a9d-96e3-83b771c7be99">

**Comment by xenova** [comment]
Hi there :wave: Which version of transformers.js are you using?

**Comment by pthieu** [comment]
@xenova `"@xenova/transformers": "^2.14.2"`

**Comment by xenova** [comment]
Right, this is because we're trying to recreate the same API as the python library. 😅 If someone maybe knows how to correctly type the `Tensor` class to support array-like indexing, then feel free to open a PR! 🤗

**Comment by xenova** [comment]
Fixed in recent Transformers.js versions.

Also, if you want to convert the tensor to a javascript object, we recommend two ways:
1. `tensor.tolist()` will convert the tensor to a multidimensional array of JS numbers
2. `tensor[0][0][0].item()` if you want to convert a scalar to a JS number
